### PR TITLE
Set tasks associated with deleted caregivee-caregiver to false

### PR DESCRIPTION
**Pull request by cassandrale179**
## Issue
- When we delete the caregivees from caregiver, we want to set assignedStatus of all the tasks created by that caregiver to be false.
- Zak's SetTask fragment used his own version of the Task object, so I refactor it to use our object.

## Solution 
- Just query the tasks associated with that caregivee and set to false. 
```java
public void setAssignedStatusForCaregivee(String caregiveeId){
    Task taskModelObject = new Task();
    taskModelObject.getAllTasks(caregiveeId, new App.TaskCallback() {
        @Override
        public void onDataReceived(List<Task> tasks){
            for (Task task : tasks){
                DatabaseReference taskRef =  database.child("users").child(caregiveeId)
                        .child("rooms").child(task.room)
                        .child("tasks").child(task.taskId);
                if (task.assignedStatus && task.caregiverId.equals(userId)){
                    task.assignedStatus = false;
                    taskRef.child("assignedStatus").setValue(false);
                }
            }
        }
    });
}
```

https://user-images.githubusercontent.com/22923895/114063914-1c5f6780-9867-11eb-8ae5-f324ef29e88e.mov

## Side Effect
- There is one side-effect I notice when testing (That I think got to do more with Android Studio than with the code) 
- What happen is: there is 2 caregiver (juigiver@gmail.com and mkgiver@gmail.com), both are caregivers of "Caregivee Jui" juigivee@gmail.com
- If we delete the relationship between juigiver and juigvee, the tasks assignedStatus are successfully set to false
- However, if another caregiver (mkgiver@gmail.com) tried to change them to true, the Firebase REFUSE to change it back, unless we press STOP on the Android app and reload it again. 
- Android Studio kept mistakenly call this `reAssignedStatus` function even though Caregiver Jui had already logged out and be replaced by Caregiver Mannika. 
- Not sure why this is happening, but I suspect it's because 2 caregivers had the same phone, it's a problem. We need to test if it is an issue if there are 2 caregivers with 2 different devices cause the same issue. 

https://user-images.githubusercontent.com/22923895/113486796-6ec51080-9482-11eb-9756-07e6cfabf59c.mov

### CODE REVIEW

File: app/src/main/java/com/example/caregiver/ArrayCheckboxAdapter.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public View getView (int position, View convertView, ViewGroup parent)
 
         // Add the descriptive text
         TextView text = new TextView(this.getContext());
-        text.setText(this.getItem(position).toString());
+        text.setText("   "  + this.getItem(position).toString());
```
Comment (cassandrale179) [inline app/src/main/java/com/example/caregiver/ArrayCheckboxAdapter.java:58]: Just adding space between the title of the task with the checkbox on Set Task page.

File: app/src/main/java/com/example/caregiver/HomeCaregiver.java [hunk 2]
```diff
@@ -147,13 +147,36 @@ public void removeCaregiveePopUp(int groupPosition){
...
+                    setAssignedStatusForCaregivee(caregiveeId);
                     userRef.child("caregivees").child(caregiveeId).removeValue();
                     ref.child(caregiveeId).child("caregivers").child(userId).removeValue();
                     startActivity(new Intent(getContext(), Dashboard.class));
                 }).setNegativeButton("No", null);
         builder.create().show();
     }
 
+    /**
+     * After caregiver deleted caregivee, all tasks assigned to them should be false.
```
Comment (cassandrale179) [inline app/src/main/java/com/example/caregiver/HomeCaregiver.java:159]: Function to set assignedStatus to false for the tasks page.

File: app/src/main/java/com/example/caregiver/SetTasksFragment.java [hunk 3]
```diff
@@ -126,46 +128,6 @@ public void onCancelled(@NonNull DatabaseError error) {
         name.addValueEventListener(valueEventListener);
     }
 
-    public class Task {
```
Comment (cassandrale179) [inline app/src/main/java/com/example/caregiver/SetTasksFragment.java:129]: I remove Zak's Task class to use our model/Task.java instead so it is uniform.

File: app/src/main/java/com/example/caregiver/HomeCaregiver.java (hunk 2 above)
Comment (Sppurswani) [reply app/src/main/java/com/example/caregiver/HomeCaregiver.java:159]: This function looks great! Could we rename it to setAssignedStatusToFalse (just so its more intuitive of what it does elsewhere)?

**Review by Sppurswani — APPROVED** [review]

**Pull request merged**